File: python/998. Maximum Binary Tree II.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def insertIntoMaxTree(self, root, val):
        """
        :type root: TreeNode
        :type val: int
        :rtype: TreeNode
        """
        node = TreeNode(val)
        if not root or val > root.val:
            node.left = root
            return node
        
        tmp = root
        while tmp.right and tmp.right.val > val:
            tmp = tmp.right
        node.left = tmp.right
        tmp.right = node
        return root

# 附加值添加到A的末尾        

File: go/src/leetcode/998/998. Maximum Binary Tree II.go
/**
@author ZhengHao Lou
Date    2022/08/30
*/
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func insertIntoMaxTree(root *TreeNode, val int) *TreeNode {
	node := &TreeNode{Val: val}
	if root == nil || val > root.Val {
		node.Left = root
		return node
	}

	tmp := root
	for tmp.Right != nil && tmp.Right.Val > val {
		tmp = tmp.Right
	}
	node.Left = tmp.Right
	tmp.Right = node
	return node
}
